### PR TITLE
feat(local-models): per-model MTP override for uncatalogued GGUFs

### LINKED ISSUE #114108: [Feature]: Set MTP status for chocolate models
### Problem or Use Case

I am using a model called "Swift-Qwen3.8-27B-Uncensored-Dynamic-MTP-UD-Q6_K_XL".   Herme's model ID catalog doesn't recognize this model, thus MTP isn't enabled.

### Proposed Solution

It would be good if we can have the ability to enable and disable features like MTP on a per-model basis.

### Alternatives Considered

_No response_

### Feature Type

Configuration option

### Scope

Medium (few files, < 300 lines)

### Contribution

- [ ] I'd like to implement this myself and submit a PR

### Debug Report (optional)

```shell

```

### PR DESCRIPTION
**Pull request by kokhlo**
Closes #114108.

## Diagnosis

MTP capability is resolved exclusively from the curated catalog — `hermes_cli/local_runtime/presets.py`:

```python
is_mtp = entry.mtp if entry is not None else model_id in mtp_capable
```

For any staged GGUF whose model id has no catalog entry, `is_mtp` falls back to the `mtp_capable` set, and the only production caller (`bootstrap._generate_presets`) never passes one — the set is always empty. Downstream, `plan_launch` never applies the MTP memory posture (kv_scale 1.2 + doubled logits buffers) and `launch_args` never emits `--spec-type draft-mtp --backend-sampling`. A dynamically-MTP'd build like the reporter's `Swift-Qwen3.8-27B-Uncensored-Dynamic-MTP-UD-Q6_K_XL` is staged and launched exactly like a plain non-MTP file, silently.

## Fix

Mirrors the existing window-override machinery (`window_overrides.json` + its consumption in `preset_for_model`):

- **`growth.py`** — `mtp_overrides.json` in `runtimes_root()` (machine-scoped, same rationale as the preset INI): `load_mtp_overrides` / `save_mtp_override` / `clear_mtp_override`. `_write_overrides` gains an optional `path` so both stores share it.
- **`presets.py`** — `preset_for_model` consults the store: an explicit override beats the catalog, the catalog stays the default, and a model the catalog has never heard of runs without spec decode unless asked — backend sampling doubles the logits buffers, so it must never turn on by guessing. The `backend-sampling`/`spec-draft-backend-sampling` keys now follow `is_mtp` itself rather than requiring a catalog entry.
- **`web_routers/local_models.py`** — `POST /api/local-models/models/{model_id}/mtp` (`{model_id, enabled}`; `enabled: null` returns to the catalog's posture) bounces the router so the new launch policy takes effect. `DELETE /models/{model_id}` also clears the override, same as the window override.

## Tests

`tests/hermes_cli/test_local_mtp_overrides.py` (solo-run like CI's file-per-process):

- no override + no catalog entry → plain launch, no spec-decode flags, no backend sampling
- override `true` on an uncatalogued model → `spec-type=draft-mtp` + both backend-sampling keys
- clearing the override returns to the catalog's verdict (off for unknown models)
- catalog MTP model + override `false` → spec decode suppressed; stock behavior unchanged without an override

RED verified by sabotaging the override consumption in `preset_for_model` (both tests fail), then GREEN on the restored fix. `ruff check` clean; hunks are formatter-stable (`ruff format --diff` touches only pre-existing drift, none of my lines).

Claim on the issue: https://github.com/NousResearch/hermes-agent/issues/114108#issuecomment-5714516626

### CODE REVIEW

**Comment by kyssta-exe** [comment]
## Review

**Summary**
Closes #114108: uncatalogued dynamically-MTP'd GGUFs (e.g. `Swift-Qwen3.8-27B`) previously launched as plain non-MTP because `is_mtp` fell back to an always-empty set. The fix mirrors the proven window-override machinery with a per-model MTP override store plus API. No CI checks reported on this branch.

**What changed**
- `hermes_cli/local_runtime/growth.py`: `mtp_overrides.json` store (`load/save/clear_mtp_override`); `_write_overrides` gains optional `path`.
- `hermes_cli/local_runtime/presets.py`: explicit override beats catalog; unknown models default to non-MTP; `backend-sampling` keys now follow `is_mtp` rather than requiring a catalog entry.
- `hermes_cli/web_routers/local_models.py`: `POST /api/local-models/models/{model_id}/mtp` (`enabled: null` clears); `DELETE` also clears the MTP override; 422 on path/body mismatch, 404 for unknown models.
- `tests/hermes_cli/test_local_mtp_overrides.py`: enable-on-unknown, clear-returns-to-catalog, disable-on-catalog-MTP.

**Strengths**
- Safe default: never guesses MTP on (backend sampling doubles logits buffers); override is explicit.
- API validation and delete-cleanup prevent stale posture leaking onto a future same-id model.
- RED→GREEN verified by sabotaging override consumption; ruff clean.

**Findings**
- One non-blocking concern: `hermes_cli/web_routers/local_models.py:736` — the save path uses `_quiet(...)` then unconditionally returns `{"ok": True, "mtp_override": ...}`. If the save fails, the caller sees success while the posture didn't change. Consider surfacing the failure (or at least returning the effective stored value).
- Polish: `presets.py:94` calls `load_mtp_overrides()` (file read) on every `preset_for_model` call; fine at this scale, but a cached load would help if preset resolution ever runs in a hot loop.

**Verdict**
Looks good to merge (consider the `_quiet`-masks-save-failure tip as follow-up).

*Reviewed using Hermes-Agent*

**Comment by kokhlo** [comment]
Thanks for the review — good catch on the save path.

Fixed in f27ccadf3f99:

- `save_mtp_override` failures now surface as a 500 with "could not persist the MTP override for <model_id>" instead of a silent `_quiet` swallow (the write *is* the point of the call — a masked failure would leave the dashboard believing the posture changed while the next start still runs the old one). While testing the old form I also noticed the `warn=` string had no `%s` placeholder, so the swallowed exception actually crashed the log formatter — both problems gone now that the save path is a plain `try/except`.
- The response now reports `growth.load_mtp_overrides().get(model_id)` — the store's effective value — rather than echoing `body.enabled`.
- Two route tests added: a roundtrip asserting the reported value matches the store (set True, then clear -> `null`), and a store-write-failure test asserting the 500 (verified RED first by restoring the old masking form, then GREEN).

Left the per-call `load_mtp_overrides()` in `preset_for_model` as-is for now, per your "fine at this scale" — happy to add a cached load if it ever shows up in a profile.
